module/zfs/zpl_file.c

```diff
@@ -798,7 +801,17 @@ zpl_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 static long
 zpl_compat_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 {
-	return (zpl_ioctl(filp, cmd, arg));
+	switch (cmd) {
+	case FS_IOC32_GETFLAGS:
+		cmd = FS_IOC_GETFLAGS;
+		break;
+	case FS_IOC32_SETFLAGS:
+		cmd = FS_IOC_SETFLAGS;
+		break;
+	default:
+		return (-ENOTTY);
```

Comment:
Interesting, I was just basing my comment of what ext2/3/4 does.  I'm glad to took a bigger sample.  Yes, I'm OK leaving this as -ENOTTY.  It can always be changed if needed in a latter commit if needed.